include/stratosphere/kvdb/kvdb_auto_buffer.hpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2018-2019 Atmosphère-NX
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include <switch.h>
+#include "../defines.hpp"
+#include "../results.hpp"
+
+namespace sts::kvdb {
+
+    class AutoBuffer {
+        NON_COPYABLE(AutoBuffer);
+        private:
+            std::unique_ptr<u8[]> buffer;
+            size_t size;
+        public:
+            AutoBuffer() : size(0) { /* ... */ }
+
+            AutoBuffer(AutoBuffer &&rhs) {
+                this->buffer = std::move(rhs.buffer);
+                this->size = rhs.size;
+                rhs.size = 0;
+            }
+
+            AutoBuffer& operator=(AutoBuffer &&rhs) {
+                rhs.Swap(*this);
+                return *this;
+            }
+
+            void Swap(AutoBuffer &rhs) {
+                std::swap(this->buffer, rhs.buffer);
+                std::swap(this->size, rhs.size);
+            }
+
+            void Reset() {
+                this->buffer.reset();
+                this->size = 0;
+            }
+
+            u8 *Get() const {
+                return this->buffer.get();
+            }
+
+            size_t GetSize() const {
+                return this->size;
+            }
+
+            Result Initialize(size_t size) {
+                /* Check that we're not already initialized. */
+                if (this->buffer != nullptr) {
+                    std::abort();
+                }
+
+                /* Allocate a buffer. */
+                auto mem = new (std::nothrow) u8[size];
```

Comment:
`std::nothrow` is a lie. Internally this is just `try { call throwing operator new; } catch { return nullptr; }`. If what you wanted (amongst other things) was avoiding bloat caused by exception handling, this won't accomplish it.